nutanix/resource_nutanix_pbr.go

```diff
@@ -0,0 +1,634 @@
+						},
+						"subnet_ip": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"prefix_length": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"destination": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"address_type": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice([]string{"INTERNET", "ALL"}, false),
+						},
+						"subnet_ip": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"prefix_length": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"protocol_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"TCP", "UDP", "ICMP", "PROTOCOL_NUMBER", "ALL"}, false),
+			},
+			"protocol_parameters": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"udp": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"source_port_range_list":      portRangeSchema(),
+									"destination_port_range_list": portRangeSchema(),
+								},
+							},
+						},
+						"tcp": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"source_port_range_list":      portRangeSchema(),
+									"destination_port_range_list": portRangeSchema(),
+								},
+							},
+						},
+						"icmp": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"icmp_type": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+									"icmp_code": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+								},
+							},
+						},
+						"protocol_number": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"action": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"DENY", "PERMIT", "REROUTE"}, false),
+			},
+			"service_ip_list": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"vpc_reference_uuid": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"api_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"metadata": {
```

Comment:
Can we attach categories to PBRs. If yes, then please include them as well.

Reply:
No categories present in PBR resource.